### PR TITLE
feat(openapi): adapt OpenAPIv2 about sheet

### PR DESCRIPTION
**Pull request by wangyuqing0424**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-27k14q0 OpenAPIv2 对接 —— Sheet](https://app.clickup.com/t/27k14q0)

File: graviti/openapi/sheet.py [hunk 1]
```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
```
Comment (graczhual) [inline graviti/openapi/sheet.py:5]: commit message: adapt OpenAPIv2 about sheet

File: graviti/openapi/sheet.py [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+from typing import Any, Dict, Optional
+from urllib.parse import urljoin
+
+from graviti.openapi.requests import open_api_do
+
+
+def _list_sheet(
+    access_key: str,
+    url: str,
+    *,
```
Comment (graczhual) [inline graviti/openapi/sheet.py:17]: do not need * here and _get_sheet

File: graviti/openapi/sheet.py [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+    offset: int,
+    limit: int,
+) -> Dict[str, Any]:
+
```
Comment (graczhual) [inline graviti/openapi/sheet.py:21]: blank line

File: graviti/openapi/sheet.py [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+def _get_sheet(
+    access_key: str,
+    url: str,
+    *,
+    schema_format: Optional[str],
+) -> Dict[str, Any]:
+    params: Dict[str, Any] = {}
+    if schema_format is not None:
+        params["schema_format"] = schema_format
```
Comment (graczhual) [inline graviti/openapi/sheet.py:35]: params = {"schema_format": schema_format}

File: graviti/openapi/sheet.py [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+    owner: str,
+    dataset: str,
+    *,
+    draft_number: int,
+    name: str,
+    schema: Dict[str, str],
+    primary_key_strategy: Optional[str] = None,
+) -> Dict[str, str]:
+    r"""Execute the OpenAPI `POST /v2/datasets/{owner}/{dataset}/drafts/{draft_number}/sheets`.
+
```
Comment (graczhual) [inline graviti/openapi/sheet.py:52]: `r`?

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+                "name": "trainval",
+                "created_at": "2021-03-05T18:58:10Z",
+                "updated_at": "2021-03-06T18:58:10Z",
+                "data_volume": 60000
+            }
+        ],
+        "offset": 0,
+        "record_size": 2,
+        "total_count": 2
+        }
```
Comment (linjiX) [inline graviti/openapi/sheet.py:200]: the indentation is wrong here.

File: graviti/openapi/sheet.py [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+        ...     "https://api.graviti.com/",
+        ...     "czhual",
+        ...     "MNIST",
+        ...     draft_number = 1,
+        ...     name = "sheet-2",
+        ...     schema = "{"portex_schema": "{\"imports\": [{\"repo\": \
+        ...         \"https://github.com/Project-OpenBytes/standard@main\", \"types\": \
+        ...         [{\"name\": \"tensor.Image\"}]}], \"type\": \"record\", \"fields\": \
+        ...         [{\"name\": \"image\", \"type\": \"tensor.Image\", \"dtype\": \"uint8\", \
+        ...         \"shape\": [28, 28]}]}","arrow_schema": ""}"
```
Comment (graczhual) [inline graviti/openapi/sheet.py:79]: schema is dict here

File: graviti/openapi/sheet.py [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+        ...     "MNIST",
+        ...     draft_number = 1,
+        ... )
+        {
+        "sheets": [
+            {
+                "name": "test",
+                "created_at": "2021-03-03T18:58:10Z",
+                "updated_at": "2021-03-04T18:58:10Z",
+                "data_volume": 10000
```
Comment (graczhual) [inline graviti/openapi/sheet.py:131]: do not add "data_volume" in List sheet(draft and commit)

File: graviti/openapi/sheet.py [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+        ...     "MNIST",
+        ...     draft_number = 1,
+        ...     sheet = "sheet-2"
+        ... )
+        {
+            "name": "trainval",
+            "created_at": "2021-03-05T18:58:10Z",
+            "updated_at": "2021-03-06T18:58:10Z",
+            "data_volume": 10000,
+            "schema": '{
```
Comment (linjiX) [inline graviti/openapi/sheet.py:247]: why the schema example is different with create_sheet?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py (hunk 9 above)
Comment (graczhual) [inline graviti/openapi/sheet.py:247]: change the schema

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py [hunk 10]
```diff
@@ -0,0 +1,342 @@
...
+    url: str,
+    owner: str,
+    dataset: str,
+    *,
+    draft_number: int,
+    name: str,
+    schema: Dict[str, str],
+    _arrow_schema: str,
+    primary_key_strategy: Optional[str] = None,
+) -> Dict[str, str]:
```
Comment (graczhual) [inline graviti/openapi/sheet.py:48]: None

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py [hunk 11]
```diff
@@ -0,0 +1,342 @@
...
+        ...     draft_number = 1,
+        ...     name = "val",
+        ...     schema = "{\"imports\": [{\"repo\":
+        ...         \"https://github.com/Project-OpenBytes/standard@main\", \"types\":
+        ...         [{\"name\": \"label.Category\"}, {\"name\": \"tensor.Image\"}]}],
+        ...         \"type\": \"record\", \"fields\": [{\"name\": \"image\", \"type\":
+        ...         \"tensor.Image\", \"dtype\": \"uint8\", \"shape\": [28, 28]}, {\"name\":
+        ...         \"category\", \"type\": \"label.Category\", \"categories\": [0, 1, 2, 3,
+        ...         4, 5, 6, 7, 8, 9]}]}",
+        ...     "_arrow_schema": ""
```
Comment (graczhual) [inline graviti/openapi/sheet.py:81]: _arrow_schema =

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py [hunk 12]
```diff
@@ -0,0 +1,342 @@
...
+
+def create_sheet(
+    access_key: str,
+    url: str,
+    owner: str,
+    dataset: str,
+    *,
+    draft_number: int,
+    name: str,
+    schema: Dict[str, str],
```
Comment (graczhual) [inline graviti/openapi/sheet.py:45]: string

File: graviti/openapi/sheet.py [hunk 13]
```diff
@@ -0,0 +1,342 @@
...
+    r"""Execute the OpenAPI `POST /v2/datasets/{owner}/{dataset}/drafts/{draft_number}/sheets`.
+
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        draft_number: The draft number.
+        name: The sheet name.
+        schema: The schema to be passed in.
```
Comment (graczhual) [inline graviti/openapi/sheet.py:58]: The portex schema of the sheet.

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py [hunk 14]
```diff
@@ -0,0 +1,342 @@
...
+        _arrow_schema: The arrow schema to be passed in.
```
Comment (graczhual) [inline graviti/openapi/sheet.py:59]: The arrow schema of the sheet.

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py [hunk 15]
```diff
@@ -0,0 +1,342 @@
...
+        ...     draft_number = 1,
+        ...     name = "val",
+        ...     schema = "{\"imports\": [{\"repo\":
+        ...         \"https://github.com/Project-OpenBytes/standard@main\", \"types\":
+        ...         [{\"name\": \"label.Category\"}, {\"name\": \"tensor.Image\"}]}],
+        ...         \"type\": \"record\", \"fields\": [{\"name\": \"image\", \"type\":
+        ...         \"tensor.Image\", \"dtype\": \"uint8\", \"shape\": [28, 28]}, {\"name\":
+        ...         \"category\", \"type\": \"label.Category\", \"categories\": [0, 1, 2, 3,
+        ...         4, 5, 6, 7, 8, 9]}]}",
+        ...     _arrow_schema: "The arrow schema of the sheet."
```
Comment (graczhual) [inline graviti/openapi/sheet.py:81]: _arrow_schema = “”

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py [hunk 16]
```diff
@@ -0,0 +1,342 @@
...
+        ... )
```
Comment (graczhual) [inline graviti/openapi/sheet.py:82]: blank line

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py [hunk 17]
```diff
@@ -0,0 +1,342 @@
...
+
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        commit_id: The commit id..
+        sheet: The sheet name.
+        schema_format: Fill "JSON"/"YAML" to determine whether the schema_format of the returned
+            schema is json or yaml. If None, it is to return json
```
Comment (graczhual) [inline graviti/openapi/sheet.py:278]: None means "JSON" format. get_draft_sheet also need to modify

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/sheet.py [hunk 18]
```diff
@@ -0,0 +1,345 @@
...
+                {
+                    "name": "trainval",
+                    "created_at": "2021-03-05T18:58:10Z",
+                    "updated_at": "2021-03-06T18:58:10Z",
+                }
+            ],
+            "offset": 0,
+            "record_size": 2,
+            "total_count": 2
+        }
```
Comment (graczhual) [inline graviti/openapi/sheet.py:144]: blank line

File: graviti/openapi/sheet.py [hunk 19]
```diff
@@ -0,0 +1,345 @@
...
+        ...     draft_number = 1,
+        ...     name = "val",
+        ...     schema = "{\"imports\": [{\"repo\":
+        ...         \"https://github.com/Project-OpenBytes/standard@main\", \"types\":
+        ...         [{\"name\": \"label.Category\"}, {\"name\": \"tensor.Image\"}]}],
+        ...         \"type\": \"record\", \"fields\": [{\"name\": \"image\", \"type\":
+        ...         \"tensor.Image\", \"dtype\": \"uint8\", \"shape\": [28, 28]}, {\"name\":
+        ...         \"category\", \"type\": \"label.Category\", \"categories\": [0, 1, 2, 3,
+        ...         4, 5, 6, 7, 8, 9]}]}",
+        ...     _arrow_schema: ""
```
Comment (graczhual) [inline graviti/openapi/sheet.py:81]: `=` not `:`

File: graviti/openapi/sheet.py [hunk 20]
```diff
@@ -0,0 +1,345 @@
...
+    """
+    url = urljoin(url, f"v2/datasets/{owner}/{dataset}/drafts/{draft_number}/sheets")
+    post_data = {"name": name, "schema": schema, "_arrow_schema": _arrow_schema}
+
+    if primary_key_strategy is not None:
+        post_data["primary_key_strategy"] = primary_key_strategy
+
+    return open_api_do(  # type: ignore[no-any-return]
+        "POST", access_key, url, json=post_data
+    ).json()
```
Comment (graczhual) [inline graviti/openapi/sheet.py:93]: return None, do not use return

File: graviti/openapi/sheet.py [hunk 21]
```diff
@@ -0,0 +1,345 @@
...
+        dataset: Name of the dataset, unique for a user.
+        draft_number: The draft number.
+        name: The sheet name.
+        schema: The portex schema of the sheet..
+        _arrow_schema: The arrow schema of the sheet.
+        primary_key_strategy: The primary key generation strategy.
+            If None, it is batch auto-increment sorting primary key.
+
+    Returns:
+        The response of OpenAPI.
```
Comment (graczhual) [inline graviti/openapi/sheet.py:64]: no Returns

**Review by graczhual — CHANGES_REQUESTED** [review]

**Review by graczhual — APPROVED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**